src/components/IncidentRequestDialog/IncidentRequestDialog.vue

```diff
@@ -0,0 +1,161 @@
+<script setup lang="ts">
+import { ref, watch } from 'vue';
+
+import { useModal } from '@/composables/useModal';
+
+import { postIncidentRequest } from '@/services/FlawService';
+import { IncidentKindEnum, IncidentKindEnumLabels, ZodIncidentRequestSchema } from '@/types/zodFlaw';
+import { useToastStore } from '@/stores/ToastStore';
+import { getDisplayedOsidbError } from '@/services/osidb-errors-helpers';
+import Modal from '@/widgets/Modal/Modal.vue';
+import LabelTextarea from '@/widgets/LabelTextarea/LabelTextarea.vue';
+import LabelSelect from '@/widgets/LabelSelect/LabelSelect.vue';
+
+const props = defineProps<{
+  flawId: string;
+  showModal: boolean;
+}>();
+
+const emit = defineEmits<{
+  hideModal: [];
+  requestSubmitted: [];
+}>();
+
+const { closeModal, isModalOpen, openModal } = useModal();
+
+const incidentReason = ref<string | undefined>('');
+const incidentKind = ref<string | undefined>(IncidentKindEnum.MAJOR_INCIDENT_REQUESTED);
+const isSubmitting = ref(false);
+const validationError = ref('');
+const submissionError = ref('');
+
+const { addToast } = useToastStore();
+
+// Options for the incident kind dropdown
+const incidentKindOptions = {
```

Comment:
```suggestion
const incidentKindOptions = IncidentKindEnumLabels.fromEntries(
  Object.entries(IncidentKindEnumLabels).map([k,v]=>[v,k])
)
```

Advantage is not having to update the component if the schema changes.